frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java

```diff
@@ -80,6 +87,38 @@ private void processMacheteGeneratedEntry(MacheteGeneratedEntry macheteEntry, An
                   .format(processedBranchName))
           .range(branch).create();
     }
+
+    // update the state of the .git/machete VirtualFile so that new entry is available in the VirtualFile
+    ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, () -> saveDocumentBeforeCheck(file));
+    // check for duplicate entries in the machete file
+    try {
+      val branchNamesFromFile = VfsUtil.loadText(file.getVirtualFile());
```

Comment:
For clarity, the purpose of `fileDocManager.saveDocument(...)` is so that `VfsUtil.loadText(...)` returns an up-to-date value here? 🤔

Reply:
Yes. Without it:
- `VfsUtil.loadText` "_doesn't know_" that there is some new entry in the file. 
- `val branchNamesFromFile` would not contain the new added entry